### PR TITLE
Created user controller

### PR DESCRIPTION
**Pull request by alanlamaswolox**
## Summary
Created user controller

## Trello card
[link](https://trello.com/c/KNixaDoN/21-crear-user-controller-con-las-operaciones-crud-de-user)

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) throws UserNotFoundException {
+        userRepository.findById(id)
+            .orElseThrow(() -> new UserNotFoundException(id.toString(), "id"));
+        userRepository.deleteById(id);
+    }
+
+    @PostMapping
+    public void create(@RequestBody User user){
+        userRepository.save(user);
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/UserController.java]: Add a user return to show the user you added at the end of the transaction

File: src/main/java/wolox/training/exceptions/UserNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "User not found")
+public class UserNotFoundException extends Exception{
+    public UserNotFoundException(final String value, final String attr) {
+        super("User with " + attr + " = " + value + " doesn't found");
+    }
+}
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/exceptions/UserNotFoundException.java]: Add an empty line to the end of the file so it doesn't cause conflicts in git.

File: src/main/java/wolox/training/models/User.java [hunk 3]
```diff
@@ -78,6 +78,9 @@ public void addBook(Book book) throws BookAlreadyOwnedException {
         this.books.add(book);
     }
 
+    public void removeBook(Book book) {
+        books.remove(book);
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/User.java:86]: Check if the book you are trying to delete exists in the book list.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public void detachBook(@PathVariable Long book_id, @PathVariable Long user_id)
+        throws UserNotFoundException, BookNotFoundException, BookAlreadyOwnedException {
+        User user = userRepository.findById(user_id)
+            .orElseThrow(() -> new UserNotFoundException(user_id.toString(), "id"));
+        Book book = bookRepository.findById(book_id)
+            .orElseThrow(() -> new BookNotFoundException(book_id.toString(), "id"));
+        user.removeBook(book);
+    }
+
+    @PostMapping("/attach/{user_id}/{book_id}")
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/UserController.java:104]: Having the same url as the previous controller can generate conflicts.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping
+    public User getUser(@PathVariable Long id) throws UserNotFoundException {
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/controllers/UserController.java:37]: Add java docs.

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Great Job 😄

File: src/main/java/wolox/training/controllers/UserController.java (hunk 4 above)
Comment (alanlamaswolox) [reply src/main/java/wolox/training/controllers/UserController.java:104]: I tried to write "/detach/blabla" jaja

**Review by AleFalcon — APPROVED** [review]
It's ok with comments by Nico

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**